crypto/asn1/a_mbstr.c

```diff
@@ -342,54 +341,8 @@ static int cpy_utf8(unsigned long value, void *arg)
     return 1;
 }
 
-/* Return 1 if the character is permitted in a PrintableString */
-static int is_printable(unsigned long value)
-{
-    int ch;
-    if (value > 0x7f)
-        return 0;
```

Comment:
Given the overall context it's not significant. This is because I can't see that input would be wider than 32 bits, so that there won't be truncations on call to ossl_isasn1print. However, one can still wonder if it makes sense to keep this comparison for better readability/auditability. But not here, at caller's. I mean caller does a number of value > X comparisons and it would probably be appropriate to explicitly denote the boundary condition in question. Rationale is following. When you read the code and note that value is unsigned long, you will inevitably ask what is ossl_isans1print exactly. Is there implicit cast and is it safe? [Even if it was more conventional iswhatever, you'd still be obliged to double-check.] But with explicit boundary condition right there in `if` at caller's, you'll be excused from going and figuring it out.

In the context it would probably be natural to get rid of is_numeric and do the check inline at caller's. Also for better readabilty/auditability.